dask/dataframe/io/delta/core.py

```diff
@@ -0,0 +1,289 @@
+                f"File {checkpoint_path} not found"
+            )
+
+        parquet_checkpoint = read_parquet(
+            checkpoint_path, engine=self.engine, storage_options=self.storage_options
+        ).compute()
+
+        # reason for this `if condition` was that FastParquetEngine seems to normalizes the json present in each column
+        # not sure whether there is a better solution for handling this .
+        # `https://fastparquet.readthedocs.io/en/latest/details.html#reading-nested-schema`
+        if self.engine.__name__ == "ArrowDatasetEngine":
+            for i, row in parquet_checkpoint.iterrows():
+                if row["metaData"] is not None:
+                    # get latest schema/columns , since delta Lake supports Schema Evolution.
+                    self.schema = schema_from_string(row["metaData"]["schemaString"])
+                elif row["add"] is not None:
+                    self.pq_files.add(f"{self.path}/{row['add']['path']}")
+
+        elif self.engine.__name__ == "FastParquetEngine":
+            for i, row in parquet_checkpoint.iterrows():
+                if row["metaData.schemaString"]:
+                    self.schema = schema_from_string(row["metaData.schemaString"])
+                elif row["add.path"]:
+                    self.pq_files.add(f"{self.path}/{row['add.path']}")
+
+    def get_pq_files_from_delta_json_logs(self):
+        """
+        start from checkpoint id, collect logs from every json file until the
+        given version
+        example:
+            checkpoint 10, version 16
+                1. read the logs from 10th checkpoint parquet ( using above func)
+                2. read logs from json files until version 16
+        log Collection:
+            for reading the particular version of delta table, We are concerned
+            about `add` and `remove` Operation (transaction) only.(which involves
+            adding and removing respective parquet file transaction)
+        """
+        log_files = self.fs.glob(
+            f"{self.delta_log_path}/{self.checkpoint // 10:019}*.json"
+        )
+        if len(log_files) == 0:
+            raise RuntimeError(
+                f"No Json files found at _delta_log_path:- {self.delta_log_path}"
+            )
+        log_files = sorted(log_files)
+        log_versions = [
+            int(re.findall(r"(\d{20})", log_file_name)[0])
+            for log_file_name in log_files
+        ]
+        if (self.version is not None) and (self.version not in log_versions):
+            raise ValueError(
+                f"Cannot time travel Delta table to version {self.version}, Available versions for given "
+                f"checkpoint {self.checkpoint} are {log_versions}"
+            )
+        for log_file_name, log_version in zip(log_files, log_versions):
+            with self.fs.open(log_file_name) as log:
+                for line in log:
```

Comment:
Again, `fs.cat` might well be faster than `open`ing the file and relying on readahead buffering.

Reply:
Got it, Thanks for the suggestion once again, changed from fs.open to fs.cat